packages/design-system/src/components/dropDown/index.js

```diff
@@ -85,12 +87,34 @@ export const DropDown = forwardRef(
     ref
   ) => {
     const internalRef = useRef();
+    const placementRef = useRef(placement);
+    const [dynamicPlacement, setDynamicPlacement] = useState(placement);
 
     const { activeOption, isOpen, normalizedOptions } = useDropDown({
       options,
       selectedValue,
     });
 
+    const positionPlacement = useCallback(
+      (popupDimensions) => {
+        // check to see if there's an overlap with the window edge
+        const neededVerticalSpace = popupDimensions?.offset.bottom;
+        // if the popup was assigned as bottom we want to always check it
+        if (
+          popupDimensions &&
+          dynamicPlacement.startsWith('bottom') &&
+          neededVerticalSpace >= window.innerHeight
+        ) {
+          setDynamicPlacement(PLACEMENT.TOP);
+        }
```

Comment:
I think this is a pretty far out edge case, but if someone were to scroll up to cause this to happen, then scroll back down would we want this to reset back to the original position? this might fix it:
```
        if (
          popupDimensions &&
          dynamicPlacement.startsWith('bottom') &&
          neededVerticalSpace >= window.innerHeight
        ) {
          setDynamicPlacement(PLACEMENT.TOP);
        } else {
          setDynamicPlacement(null);
        }
        
        ....
        
         placement={dynamicPlacement || placement}
```

example of lil edge case:
https://user-images.githubusercontent.com/35983235/133857060-0ff1c29c-5958-4570-982f-1a5eed1e0625.mp4

Reply:
Ok sorry this took a minute. I couldn't pass `null` as that would lead to an infinite loop. I realized also that by grabbing `popupDimensions` from the `onPositionUpdate` callback, I was getting stale values. So instead I am passing in the `popupRef` itself in order to calculate the popup dimension here.